Haugedal/fiks regn ki univar

haugedal opened the pull request.

haugedal committed:
Skrive i dokumentasjonen om korleis NA-verdiar skal handterast

haugedal committed:
Test på at regn_ki_univar() gjev forventa resultat

haugedal committed:
Test på at regn_ki_univar() funkar med grupperte inndata

haugedal committed:
Test på at regn_ki_univar() ignorerer NA-verdiar i inndata

haugedal committed:
Test for regn_ki_univar(bootstrap = TRUE) med NA-verdiar i inndata

Testar at regn_ki_univar() med bootstrapping ikkje gjev feilmelding
med NA-verdiar i inndata.

haugedal committed:
Fjerna dobbelt mellomrom

haugedal committed:
Ignorer NA-verdiar

haugedal committed:
Fjerna hardkoda tal på bootstrap-replikasjonar

haugedal committed:
Fjerna unødvendige na.rm = TRUE

Alle NA-verdiar vert no fjerna heilt i starten av funksjonen,
så alle na.rm = TRUE er overflødige.

statkarl commented on tests/testthat/test-ymse-funksjoner.R:
```diff
@@ -0,0 +1,11 @@
+    mean = 5.843333333333333712289459072053432464599609375,
+    high = 5.97693418515997887396906662615947425365447998046875
+  )
+  expect_true(all(sepal_length_resultat == sepal_length_forventa))
```
Her bør du vel ha `expect_identical()`?

haugedal replied:
Einig.

statkarl commented on tests/testthat/test-ymse-funksjoner.R:
```diff
@@ -9,3 +9,30 @@ test_that("regn_ki_univar() gjev forventa resultat", {
+      6.76871498088249712310471295495517551898956298828125
+    )
+  )
+  expect_true(
```
Igjen bør det vera `expect_identical()`.

haugedal replied:
Einig.

statkarl commented on tests/testthat/test-ymse-funksjoner.R:
```diff
@@ -36,3 +36,22 @@ test_that("regn_ki_univar() funkar med grupperte inndata", {
+test_that("regn_ki_univar() ignorerer NA-verdiar i inndata", {
+  expect_identical(
+    object = regn_ki_univar(alle_na),
+    expected = , regn_ki_univar(numeric())
```
Kva gjer kommaa etter `=` her (og på dei neste linjene)?

haugedal replied:
Dei stamma truleg frå at eg gjekk frå éi linje, utan namn på argumenta, til fleire linjer, og fekk med eit ekstra komma. Overraskande nok fungerte koden slik han stod, med kommaa, som me såg på.

statkarl commented on R/ymse-funksjoner.R:
```diff
@@ -191,7 +195,7 @@ regn_ki_bin = function(x, n, alfa = 0.05) {
 #'   group_by(am) %>%
 #'   summarise(regn_ki_univar(mpg))
-regn_ki_univar = function(x, bootstrap = FALSE, alfa = 0.05) {
+regn_ki_univar = function(x, bootstrap = FALSE, alfa = 0.05, R = 9999) {
```
Må ha testar på dette.

haugedal replied:
La til nokre testar for bootstrapping, og éin for val av tal på replikasjonar, `R`. Kommenter gjerne viss det er fleire testar du ser behov for.

haugedal committed:
Fjern namn på vektor

statkarl reviewed (changes requested):
Her er mine kommentarar.

statkarl reviewed (changes requested):
Her er mine kommentarar.

haugedal committed:
Flytta test inn i rett fil

haugedal committed:
Flytta delar av test inn i rett fil

Den gamle testen var i feil fil, og testa fleire ting samtidig.
Han hadde òg mean som ein namngjeven vektor, som me ikkje vil ha.

haugedal committed:
Fleire testar av regn_ki_univar() med bootstrapping

haugedal committed:
Fjerna simpleboot frå lista over importerte pakkar

Pakken er ikkje lenger i bruk i rapwhale.

haugedal committed:
Retta variabelnamn i ein test

Variabelnamna hang igjen frå førre testen,
som handa om alfa. Gjeldande test går på replikasjonar.

statkarl commented on tests/testthat/test-ymse-funksjoner.R:
```diff
@@ -0,0 +1,136 @@
+test_that("regn_ki_univar() ignorerer NA-verdiar i inndata", {
+  expect_identical(
+    object = regn_ki_univar(alle_na),
+    expected = regn_ki_univar(numeric())
```
Er fornuftig å samanlikna med `regn_ki_univar(numeric())`, men me manglar ein test på kva `regn_ki_univar(numeric())` faktisk skal gje ut. Det minste me testar, er ein vektor med lengd 1. Så det må inn ein test på dette (gjerne i testen som står øvst i fila).

haugedal replied:
Utvida den øvste testen med dette eksempelet, slik du føreslo.

statkarl reviewed (changes requested):
Berre ei lita endring til, så er denne klar.

haugedal committed:
Bytt argumentet alfa til konfidensnivå i regn_ki_univar()

haugedal committed:
Utvida test med inndata utan nokre element

statkarl reviewed (approved):
Ser bra ut.
